main/actions/src/main/scala/sbt/ForkTests.scala

```diff
@@ -73,6 +74,8 @@ private[sbt] object ForkTests {
             os.flush()
 
             new React(is, os, log, opts.testListeners, resultsAcc).react()
+          } catch {
+            case NonFatal(e) => resultsAcc("Forked test harness failed: " + e.getMessage) = SuiteResult.Error
```

Comment:
`println(e.getMessage)` is something of an antipattern, the exception class name itself is important (the message of a `NullPointerException` is typically itself `null`), as is the stack trace.

What exception are you catching here, the one thrown on the client or the deserialized, warpping`ForkError` on the server?

It would be good to test that this still would have fixed my original problem, when the test harness threw a `LinkageError` (which is not considered `NonFatal`).